_pytest/logging.py

```diff
@@ -489,14 +490,20 @@ def set_when(self, when):
         """Prepares for the given test phase (setup/call/teardown)"""
         self._when = when
         self._section_name_shown = False
+        if when == 'call':
+            self._test_outcome_written = False
 
     def emit(self, record):
         if self.capture_manager is not None:
             self.capture_manager.suspend_global_capture()
         try:
-            if not self._first_record_emitted or self._when in ('teardown', 'finish'):
+            if not self._first_record_emitted:
                 self.stream.write('\n')
                 self._first_record_emitted = True
+            elif self._when in ('teardown', 'finish'):
+                if self._test_outcome_written is False:
```

Comment:
It is considered more Pythonic to write `if not self._test_outcome_written` instead.

Reply:
But less explicit. Personal habits...

 I'll change.